Cognite.Simulator.Tests/UtilsTests/ModelLibraryExtDepsUnitTest.cs

```diff
@@ -0,0 +1,191 @@
+                ""name"": ""{f.name}"",
+                ""mimeType"": ""{f.mimeType}"",
+                ""dataSetId"": 123,
+                ""uploaded"": true
+            }}"));
+
+            return OkItemsResponse(items);
+        }
+
+
+        private static HttpResponseMessage MockSimulatorModelRevEndpoint()
+        {
+            var externalDependencies = Enumerable.Range(1, 2)
+                .Select(i => $@"{{
+                    ""file"": {{ ""id"": {100 + i} }},
+                    ""arguments"": {{ ""address"": ""test.address.{i}"" }}
+                }}").ToList();
+
+            var item = $@"{{
+                ""id"": 1234567890,
+                ""externalId"": ""TestModelExternalId-v1"",
+                ""name"": ""Test Model Revision"",
+                ""description"": ""Test model revision description"",
+                ""simulatorExternalId"": ""{SeedData.TestSimulatorExternalId}"",
+                ""modelExternalId"": ""TestModelExternalId"",
+                ""externalDependencies"": [
+                    {string.Join(",", externalDependencies)}
+                ],
+                ""fileId"": 100,
+                ""createdByUserId"": ""n/a"",
+                ""status"": ""unknown"",
+                ""dataSetId"": 123,
+                ""versionNumber"": 1,
+                ""logId"": 1234567890,
+                ""createdTime"": 1234567890000,
+                ""lastUpdatedTime"": 1234567890000
+            }}";
+            return OkItemsResponse(item);
+        }
+        private static readonly IList<SimpleRequestMocker> endpointMockTemplates = new List<SimpleRequestMocker>
+        {
+            new SimpleRequestMocker(uri => uri.EndsWith("/token"), MockAzureAADTokenEndpoint),
+            new SimpleRequestMocker(uri => uri.Contains("/simulators/models/revisions/list"), MockSimulatorModelRevEndpoint, 1),
+            new SimpleRequestMocker(uri => uri.Contains("/simulators/models/revisions/byids"), MockSimulatorModelRevEndpoint, 5),
+            new SimpleRequestMocker(uri => uri.Contains("/simulators/models/revisions/update"), MockSimulatorModelRevEndpoint, 1),
+            new SimpleRequestMocker(uri => uri.Contains("/files/byids"), MockFilesByIdsEndpoint, 1),
+            new SimpleRequestMocker(uri => uri.Contains("/files/downloadlink"), MockFilesDownloadLinkEndpoint, 3),
+            new SimpleRequestMocker(uri => uri.Contains("/files/download"), () => MockFilesDownloadEndpoint(1), 3),
+            new SimpleRequestMocker(uri => true, GoneResponse).ShouldBeCalled(Times.AtMost(100)) // doesn't matter for the test
+        };
+
+
+        /// <summary>
+        /// This test verifies that the ModelLibrary can handle external dependencies correctly.
+        /// It checks that the model revision is fetched, files are downloaded, and model is processed correctly.
+        /// </summary>
+        [Fact]
+        public async Task TestModelLibraryWithExternalDependencies()
```

Comment:
any chance we could split this up for better readability? like the setup part at least?